### ISSUE TITLE
Add RDM support

### ISSUE THREAD

**Issue by CShark** [CShark/usb_dmx#3]
Add support to manage RDM devices using ArtNET

**Comment by 517002650** [comment]
I don't know why RDM cannot be used, and the reset button cannot reset all parameters. I can only use the OLED screen to flip pages,

**Comment by 517002650** [comment]
I have processed both the PCB and board, and purchased and soldered the chip components. All other functions are normal, but the RDM cannot be used. The reset button cannot reset the configuration parameters back to default. I did not purchase and soldered the D3TVS components. Does the RDM function require soldering of the D3TVS components? I am very puzzled? There is another question. Is it possible for this chip to support direct connection with RZ45 Ethernet cable? USB virtual network card cannot achieve 8XL output

**Comment by CShark** [comment]
Reset works for me. It is initiated either using the OLED-Screen or pressing and holding the reset button for at least 5s. But yeah, there is no visual confirmation, which would probably be a good thing to have... #8

**Comment by CShark** [comment]
RDM is not yet implemented, thats why it won't work. The TVS-Diodes are not necessary for a working pcb and have nothing to do with RDM.

Currently I've already done a refactoring to allow for easy integration of the RDM functionality, but my main problem is that I don't have any device that supports RDM. I'd love to have it, but I currently have nothing solid to develop against. I could use a couple of Arduinos with some sketchy RDM, but I would prefer to have proper devices for testing.

I'll see if I can get some and push this feature a bit more in the next couple months.

Regarding Ethernet: That is currently out of scope in the near future. If I do ethernet, I'll either have to change the formfactor to accomodate the ethernet port and electronics or sacrifice a DMX port. And if I add Ethernet I'd also like to add PoE while I'm at it; I'm playing with the thought of building an alternate version but very probably not in the foreseeable future.

**Comment by 517002650** [comment]
> RDM is not yet implemented, thats why it won't work. The TVS-Diodes are not necessary for a working pcb and have nothing to do with RDM.
> 
> Currently I've already done a refactoring to allow for easy integration of the RDM functionality, but my main problem is that I don't have any device that supports RDM. I'd love to have it, but I currently have nothing solid to develop against. I could use a couple of Arduinos with some sketchy RDM, but I would prefer to have proper devices for testing.
> 
> I'll see if I can get some and push this feature a bit more in the next couple months.
> 
> Regarding Ethernet: That is currently out of scope in the near future. If I do ethernet, I'll either have to change the formfactor to accomodate the ethernet port and electronics or sacrifice a DMX port. And if I add Ethernet I'd also like to add PoE while I'm at it; I'm playing with the thought of building an alternate version but very probably not in the foreseeable future.

Yes, the reset button is useful. You are right, you need to press and hold for about 5 seconds. I am currently encountering a problem where I am unable to modify and replace the host IP properly. After modifying and replacing the host IP, I am unable to obtain the IP address properly and need to reset repeatedly. Occasionally, I can successfully modify the IP. Why is the success rate of modifying the IP so low?, Will this issue still be fixed in the future, or will it be a problem with my operation?

**Comment by 517002650** [comment]
> RDM is not yet implemented, thats why it won't work. The TVS-Diodes are not necessary for a working pcb and have nothing to do with RDM.
> 
> Currently I've already done a refactoring to allow for easy integration of the RDM functionality, but my main problem is that I don't have any device that supports RDM. I'd love to have it, but I currently have nothing solid to develop against. I could use a couple of Arduinos with some sketchy RDM, but I would prefer to have proper devices for testing.
> 
> I'll see if I can get some and push this feature a bit more in the next couple months.
> 
> Regarding Ethernet: That is currently out of scope in the near future. If I do ethernet, I'll either have to change the formfactor to accomodate the ethernet port and electronics or sacrifice a DMX port. And if I add Ethernet I'd also like to add PoE while I'm at it; I'm playing with the thought of building an alternate version but very probably not in the foreseeable future.

I am looking forward to launching RDM support. Can you provide me with an address so that I can send you some sections that support RDM for testing? I also hope to fix the issue of not being able to automatically obtain IP addresses after modifying the host IP, which is very unstable and has a low success rate.

**Comment by CShark** [comment]
I do now have a device that supports RDM (just a stupid cheap DMX to LED controller), my plan is to hopefully start on the enumeration part this month.

**Comment by 517002650** [comment]
> 添加对使用 ArtNET 管理 RDM 设备的支持



> 添加对使用 ArtNET 管理 RDM 设备的支持



> Add support to manage RDM devices using ArtNET
Does version 2.1 of V5 support RDM?

**Comment by CShark** [comment]
@517002650 No, not yet. It'll be in the next version.
